core/batcher.go

```diff
@@ -0,0 +1,348 @@
+		BatcherRequestType BatcherRequestType `json:"batcher_request_type"` // BatcherRequestType for choose method to handling request. for example tx, swaps, swaps_keys, multi_swaps, multi_swaps_keys
+	}
+
+	Batcher struct {
+		batchCacheStub *cachestub.BatchCacheStub
+		cc             *ChainCode
+		cfgBytes       []byte
+		ski            string
+		tracingHandler *telemetry.TracingHandler
+	}
+)
+
+// BatcherHandler allow execute few subtransaction (batch request) in single transaction in hlf using cache state between requests to solbe mcc problem
+// The arguments of this method contains array of requests for execution each request contain own arguments for call method
+func BatcherHandler(
+	traceCtx telemetry.TraceContext,
+	stub shim.ChaincodeStubInterface,
+	cfgBytes []byte,
+	creatorSKI, hashedCert [32]byte,
+	arguments []string,
+	cc *ChainCode,
+) ([]byte, error) {
+	tracingHandler := cc.contract.TracingHandler()
+	traceCtx, span := tracingHandler.StartNewSpan(traceCtx, BatcherBatchExecute)
+	defer span.End()
+
+	logger := Logger()
+	batchID := stub.GetTxID()
+	span.SetAttributes(attribute.String("batch_tx_id", batchID))
+	start := time.Now()
+	defer func() {
+		logger.Infof("batch %s elapsed time %d ms", batchID, time.Since(start).Milliseconds())
+	}()
+
+	if len(arguments) != 1 {
+		return nil, fmt.Errorf("expected 1 argument, got %d", len(arguments))
+	}
+
+	var batchDTO BatcherBatchExecuteRequestDTO
+	if err := json.Unmarshal([]byte(arguments[0]), &batchDTO); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal BatcherBatchExecuteRequestDTO: %w", err)
+	}
+
+	batcher, err := NewBatcher(stub, cfgBytes, cc, tracingHandler)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create batchInsertHandler: %w", err)
+	}
+
+	if err = batcher.ValidateCreator(creatorSKI, hashedCert); err != nil {
+		return nil, fmt.Errorf("failed to validate creator: %w", err)
+	}
+
+	batchResponse, batchEvent, err := batcher.HandleBatch(traceCtx, batchDTO.Requests)
+	if err != nil {
+		return nil, fmt.Errorf("failed handling batch: %w", err)
+	}
+
+	eventData, err := pb.Marshal(batchEvent)
+	if err != nil {
+		logger.Errorf("Couldn't marshal batch event %s: %s", batchID, err.Error())
+		span.SetStatus(codes.Error, "marshalling batch event failed")
+
+		return nil, fmt.Errorf("failed to marshal batch event: %w", err)
+	}
+	if err = stub.SetEvent(BatcherBatchExecuteEvent, eventData); err != nil {
+		logger.Errorf("Couldn't set batch event %s: %s", batchID, err.Error())
+		span.SetStatus(codes.Error, "set batch event failed")
+
+		return nil, fmt.Errorf("failed set batch event %s: %w", batchID, err)
+	}
+
+	data, err := pb.Marshal(batchResponse)
+	if err != nil {
+		logger.Errorf("Couldn't marshal batch response %s: %s", batchID, err.Error())
+		span.SetStatus(codes.Error, "marshalling batch response failed")
+
+		return nil, fmt.Errorf("failed marshalling batch response %s: %w", batchID, err)
+	}
+
+	return data, nil
+}
+
+const skiPattern = "^[0-9a-f]+$"
```

Comment:
The `const skiPattern = "^[0-9a-f]+$"` is only used in the `validateBatcherSKI` function. Why is it in the global scope?

Reply:
@muzykantov done, validation SKI removed